File: Data Structures & Algorithms/majority-element/submission-10.java
class Solution {
    public int majorityElement(int[] nums) {
        int result = 0;
        int count = 0;
        for(int n : nums){
            if(count == 0){
                result = n;
            }
            
            count += (n == result? 1:-1);
        }

        return result;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freq = [[] for i in range(len(nums)+1)]

        for num in nums:
            count[num] = count.get(num, 0) + 1

        for key, v in count.items():
            freq[v].append(key)
        
        res = []
        for i in range(len(freq)-1, 0,-1):
            for element in freq[i]:
                res.append(element)
                if len(res) == k:
                    return res